Dockerfile

```diff
@@ -1,46 +1,48 @@
 # Base stage for building gems
-FROM        ruby:2.5.5-stretch as base
-ENV         BUNDLER_VERSION 2.0.2
+FROM        ruby:2.5.5-stretch as bundle
 RUN         echo "deb http://deb.debian.org/debian stretch-backports main" >> /etc/apt/sources.list \
          && apt-get update && apt-get upgrade -y build-essential \
          && apt-get install -y --no-install-recommends \
             cmake \
             pkg-config \
             zip \
             git \
-         && gem install bundler \
          && rm -rf /var/lib/apt/lists/* \
          && apt-get clean
 
-RUN         mkdir -p /home/app/avalon
-WORKDIR     /home/app/avalon
-
 COPY        Gemfile ./Gemfile
 COPY        Gemfile.lock ./Gemfile.lock
-RUN         bundle config build.nokogiri --use-system-libraries \
-         && bundle install --with aws development test postgres --without production 
+
+RUN         gem install bundler -v "$(grep -A 1 "BUNDLED WITH" Gemfile.lock | tail -n 1)" \
+         && bundle config build.nokogiri --use-system-libraries
 
 
-# Download stage takes advantage of parallel build
+# Build development gems
+FROM        bundle as bundle-dev
+RUN         bundle install --with aws development test postgres --without production 
+
+
+# Download binaries in parallel
 FROM        ruby:2.5.5-stretch as download
+RUN         curl -L https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz | tar xvz -C /usr/bin/
 RUN         curl https://chromedriver.storage.googleapis.com/2.46/chromedriver_linux64.zip -o /usr/local/bin/chromedriver \
-         && chmod +x /usr/local/bin/chromedriver \
-         && curl -L https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz | tar xvz -C /usr/bin/ \
-         && mkdir -p /tmp/ffmpeg && cd /tmp/ffmpeg \
+         && chmod +x /usr/local/bin/chromedriver
+RUN         curl https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb -o /chrome.deb
+RUN         mkdir -p /tmp/ffmpeg && cd /tmp/ffmpeg \
          && curl https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.tar.xz | tar xJ \
          && cp `find . -type f -executable` /usr/bin/
 
 
-# Dev stage for building dev image
-FROM        ruby:2.5.5-slim-stretch as dev
+# Base stage for building final images
+FROM        ruby:2.5.5-slim-stretch as base
 ENV         BUNDLER_VERSION 2.0.2
```

Comment:
Should this be generalized like: https://github.com/avalonmediasystem/avalon/pull/4053/files#diff-3254677a7917c6c01f55212f86c57fbfR16

Reply:
True